ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/tasks/PipelinedImportChainSegmentTask.java

```diff
@@ -233,25 +233,30 @@ public void completeChunkPipelineAndMaybeLaunchNextOne(
     return executeWorkerSubTask(
         ethContext.getScheduler(),
         () -> {
-          final CompletableFuture<List<BlockHeader>> result = new CompletableFuture<>();
-          final BlockHeader parentHeader = headers.get(0);
-          final BlockHeader childHeader = headers.get(1);
-          final ProtocolSpec<C> protocolSpec =
-              protocolSchedule.getByBlockNumber(childHeader.getNumber());
-          final BlockHeaderValidator<C> blockHeaderValidator =
-              protocolSpec.getBlockHeaderValidator();
-          if (blockHeaderValidator.validateHeader(
-              childHeader, parentHeader, protocolContext, headerValidationMode)) {
-            // The first header will be imported by the previous request range.
-            result.complete(headers.subList(1, headers.size()));
-          } else {
-            result.completeExceptionally(
-                new InvalidBlockException(
-                    "Provided first header does not connect to last header.",
-                    parentHeader.getNumber(),
-                    parentHeader.getHash()));
+          try {
+            final CompletableFuture<List<BlockHeader>> result = new CompletableFuture<>();
+            final BlockHeader parentHeader = headers.get(0);
+            final BlockHeader childHeader = headers.get(1);
+            final ProtocolSpec<C> protocolSpec =
+                protocolSchedule.getByBlockNumber(childHeader.getNumber());
+            final BlockHeaderValidator<C> blockHeaderValidator =
+                protocolSpec.getBlockHeaderValidator();
+            if (blockHeaderValidator.validateHeader(
+                childHeader, parentHeader, protocolContext, headerValidationMode)) {
+              // The first header will be imported by the previous request range.
+              result.complete(headers.subList(1, headers.size()));
+            } else {
+              result.completeExceptionally(
+                  new InvalidBlockException(
+                      "Provided first header does not connect to last header.",
+                      parentHeader.getNumber(),
+                      parentHeader.getHash()));
+            }
+            return result;
+          } catch (Throwable t) {
```

Comment:
Curious - why do you need this try / catch?

Reply:
Oh we shouldn't need it and I've removed it again.  That said, somewhere we're swallowing exceptions silently so will need to track that down and fix separately.